### PR TITLE
fix: PL related issues in one rule and backport CVE-2023-38199 fix

### PR DESCRIPTION
**Pull request by fzipi**
Fixes:
- PL related issues in rule 921422
- Backport fix for CVE-2023-38199 in new rule 920620
- Move auditLogPart if it is in a chained rules where needed

Additional changes:
- Cleaning up PL1 tags
- Move testing framework from ftw to go-ftw
- Backport yaml test files fixes to support go-ftw
- Update copyright year
- Update sponsors list

### CODE REVIEW

**Comment by RedXanadu** [comment]
@fzipi Something's not quite right. We've got a new, rogue crs-setup.conf file `rules/crs-setup.conf.example` which managed to sneak in in `60bdfef`.

**Comment by fzipi** [comment]
Ugh, ok, removing it! Thanks for catching this one.

**Comment by RedXanadu** [comment]
@fzipi We need a new CHANGES.MD.

I'm using your change list from this PR as the basis for the blog post, currently.

**Comment by fzipi** [comment]
@RedXanadu I don't know if that should come from the dev branch. I think this is better to create a PR to the master branch directly, along with the version updates. That will be the release. Then we merge back to dev.

**Review by theseion — DISMISSED** [review]

File: crs-setup.conf.example [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # ------------------------------------------------------------------------
-# OWASP ModSecurity Core Rule Set ver.3.3.4
+# OWASP ModSecurity Core Rule Set ver.3.3.5-dev
```
[the remark this answers is not in the document]
Comment (fzipi) [reply crs-setup.conf.example:2]: That's one of the things we discussed here. The dev branch must have `-dev`. Once we merge, then we create one specific PR for master that has the correct version.
Once merged we merge back to dev, and change again to `-dev`. I don't see a better way of doing this provided the text is in the actual files.

Also documented this in the [release procedure](https://github.com/coreruleset/coreruleset/wiki/Release-Procedure). If there are better ideas, they are welcome.

**Review by RedXanadu — APPROVED** [review]
After scrolling through the changes page by page, everything looks good!

Great work. I think this PR is ready to fly :rocket:

**Review by theseion — APPROVED** [review]

**Pull request merged**